workshops/chrome-extension/popup.js

```diff
@@ -0,0 +1,98 @@
+// create global timer constiable for timer
+let timer;
+
+// communication port to background script
+const port = chrome.extension.connect({
+  name: 'Communication'
+});
+
+// set timer once start is clicked
+async function setTimer(timerSeconds) {
+  let start = new Date().getTime();
+  let end = new Date(start + timerSeconds * 1000).getTime();
+  await chrome.storage.sync.set({ endTime: end });
+  await chrome.storage.sync.set({ startTime: start });
+  port.postMessage(true);
+  updatePopUpContents(start, end);
+}
+
+// reset timer values in storage
+async function resetTimer() {
+  await chrome.storage.sync.set({ endTime: null });
+  await chrome.storage.sync.set({ startTime: null });
+  port.postMessage(true);
+  updatePopUpContents();
+}
+
+// build message stating time left in dd:hh:mm:ss format
+function getTimeLeft(endTime) {
+  const timeInEpochUntilEnd = new Date(endTime) - new Date();
+  if (timeInEpochUntilEnd < 0) {
+    return "TIME'S UP!";
+  }
+  message = formatTimer(timeInEpochUntilEnd);
+  return message + ' Left!';
+}
+
+// format countdown timer in dd:hh:mm:ss format
+function formatTimer(timeInEpochUntilEnd) {
+  // Time calculations for days, hours, minutes and seconds
+  const days = Math.floor(timeInEpochUntilEnd / (1000 * 60 * 60 * 24));
+  const hours = Math.floor((timeInEpochUntilEnd % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+  const minutes = Math.floor((timeInEpochUntilEnd % (1000 * 60 * 60)) / (1000 * 60));
+  const seconds = Math.floor((timeInEpochUntilEnd % (1000 * 60)) / 1000);
+  let message = '';
+  if (days > 0) message += days === 0 ? `00:` : days < 10 ? `0${days}:` : `${days}:`;
+  if (hours >= 0) message += hours === 0 ? `00:` : hours < 10 ? `0${hours}:` : `${hours}:`;
+  if (minutes >= 0) message += minutes === 0 ? `00:` : minutes < 10 ? `0${minutes}:` : `${minutes}:`;
+  if (seconds >= 0) message += seconds === 0 ? `00` : seconds < 10 ? `0${seconds}` : `${seconds}`;
+
+  return message;
+}
+
+// re-usable chrome storage API get request
```

Comment:
"re-usable" is not descriptive.  I think what you have done is made this a Promise instead of callback?  In which case that should be the comment.